cmd/node/node.go

```diff
@@ -689,20 +695,30 @@ func (app *SpacemeshApp) startAPIServices(postClient api.PostAPI, net api.Networ
 	// gateway server is enabled without enabling at least one GRPC service.
 
 	// Make sure we only start the server once
-	startService := func(svc grpcserver.ServiceAPI) {
+	name := app.nodeID.ShortString()
+	lg := log.NewDefault(name).WithFields(app.nodeID)
```

Comment:
the method that calls `startAPIServices` (= `Start`) is creating a logger that way already. it will be better to pass it down to this instead of creating a new one. though if you see a lot of clashes in apis and other usages than we can live with this duplication. though I'm pretty sure we only use it once so should be easy to pass down the root logger..

Reply:
Ok, makes sense. I will reuse the app.Logger I thought by using the same initial constructed one would not work. Thanks for the valuable insight.

Reply:
> Btw, I have a proposal, instead of relaying on that switch stmt why won't we just have a map of these keys like `map[keys]bool` or `map[keys]struct{}` and if an element is found then it will return ok and we can go and use the **logger**. This way we only have to modify in one place the **map** and everything will work and also we also have the benefit at runtime to add dynamic support for _future keys_.
> 
> Do you want me in this **PR** to add this or make a new one after this is merged and continue further improving this ?

I don't have an opinion on replacing the switch with a map, will defer to @y0sher and @noamnelke on that. But if we do make that change, let's do it in a separate branch/PR - small PRs are good! :)

Reply:
Ok, sounds good to me !